src/fivetran-destination/src/destination.rs

```diff
@@ -34,48 +36,167 @@ impl Destination for MaterializeDestination {
     }
 
     async fn test(&self, request: Request<TestRequest>) -> Result<Response<TestResponse>, Status> {
-        to_grpc(config::handle_test_request(request.into_inner()).await)
+        let request = request.into_inner();
+        let result = with_retry_and_logging(|| async {
+            config::handle_test_request(request.clone())
+                .await
+                .context("handle_test_request")
+        })
+        .await;
+
+        let response = match result {
+            Ok(()) => fivetran_sdk::test_response::Response::Success(true),
+            Err(e) => fivetran_sdk::test_response::Response::Failure(e.to_string()),
+        };
+        to_grpc(Ok(TestResponse {
+            response: Some(response),
+        }))
```

Comment:
note: Fivetran asked that we never return gRPC errors, that everything response with a 200 and the error embedded in that message. Which is why we always return `Ok(...)` in the handlers